src/components/posts/utils/fetchPosts.ts

```diff
@@ -0,0 +1,11 @@
+export const fetchPosts = async (index: number) => {
+  try {
+    const response = await fetch(
+      `http://localhost:3000/posts?_start=${index}&_limit=10`
+    );
+    const posts = await response.json();
```

Comment:
return response.json() powinno wystarczyc, to i tak zwraca promisa, a wydaje mi sie, ze nie ma tutaj potrzeby tworzyc dodatkowej zmiennej